include/deal.II/grid/grid_tools.h

```diff
@@ -1305,6 +1305,107 @@ namespace GridTools
+     * be converted to
+     * GridTools::internal::DistributedComputePointLocationsInternal, which can
+     * be used to reinit Utilities::MPI::RemotePointEvaluation.
+     */
+    template <int structdim, int spacedim>
+    struct DistributedComputeIntersectionLocationsInternal
+    {
+      /**
+       * Intersections are assumed to be simplices (as, e.g., provided by CGAL)
+       */
+      using IntersectionType =
+        std::array<dealii::Point<spacedim>, structdim + 1>;
+
+      /**
+       * Information of each intersection on sending/evaluation side. The
+       * elements of the tuple are as follows: 0) cell level and index, 1) rank
+       * of the owning process, 2) local index of the owning process, 3)
+       * found intersection.
+       *
+       * @note The vector is sorted according to 1), 2).
+       */
+      std::vector<std::tuple<std::pair<int, int>,
+                             unsigned int,
+                             unsigned int,
+                             IntersectionType>>
+        send_components;
+
+      /**
+       * Information of each received data value. The elements of the tuple are
+       * as follows: 0) rank of sender, 1) local index, 2) found intersections.
+       *
+       * @note The vector is sorted according to 1), 0), 2).
+       *
+       * @note Multiple intersections between cells can be found
+       */
+      std::vector<std::tuple<unsigned int, unsigned int, IntersectionType>>
+        recv_components;
+
+      /**
+       * Pointers of ranges to found intersections for requested intersection.
+       */
+      std::vector<unsigned int> recv_ptrs;
+
+      /**
+       * Distribute quadrature points to found intersections and construct
+       * GridTools::internal::DistributedComputePointLocationsInternal from
+       * class members. This can be done without searching for points again
+       * since all information is locally known.
+       *
+       * @p consistent_numbering_of_sender_and_receiver can be used to ensure
+       * points on sender and receiver side are numbered consistently.
+       * This parameter is optional if DistributedComputePointLocationsInternal
+       * is used to setup RemotePointEvaluation, but might be helpful for
+       * debugging or other usage of DistributedComputePointLocationsInternal.
+       * Note that setting this parameter true requires an additional
+       * communication during the setup phase.
+       */
+      template <int dim>
+      GridTools::internal::DistributedComputePointLocationsInternal<dim,
+                                                                    spacedim>
+      convert_to_distributed_compute_point_locations_internal(
+        const unsigned int                  n_quadrature_pnts,
+        const Triangulation<dim, spacedim> &tria,
+        const Mapping<dim, spacedim> &      mapping,
+        const bool consistent_numbering_of_sender_and_receiver = false) const;
```

Comment:
Is this parameter tested?

Reply:
Yes, in `tests/remote_point_evaluation/convert_intersections_to_rpe_data_01.cc` I set it `true` :)